### PR TITLE
fix: Model cost calculator fails with NPE for some adapters #201

### PR DESCRIPTION
**Pull request by astsiapanay**
Please read @adubovik comment:
https://github.com/epam/ai-dial-core/issues/201#issuecomment-1970746006

### CODE REVIEW

File: src/test/java/com/epam/aidial/core/util/ModelCostCalculatorTest.java [hunk 1]
```diff
@@ -61,4 +63,255 @@ public void testCalculate_TokenCost() {
...
+        Model model = new Model();
+        model.setType(ModelType.CHAT);
+        Pricing pricing = new Pricing();
+        pricing.setPrompt("0.1");
+        pricing.setCompletion("0.5");
+        pricing.setUnit("char_without_whitespace");
+        model.setPricing(pricing);
+        when(context.getDeployment()).thenReturn(model);
+
+        String response = """
```
Comment (adubovik) [inline src/test/java/com/epam/aidial/core/util/ModelCostCalculatorTest.java:134]:
```java
        String response = """
                {"error": { "message": "message", "type": "type", "param": "param", "code": "code" } } 
                """;
```

File: src/test/java/com/epam/aidial/core/util/ModelCostCalculatorTest.java [hunk 2]
```diff
@@ -61,4 +63,255 @@ public void testCalculate_TokenCost() {
...
+                data:   {"id":"chatcmpl-7VfCSOSOS1gYQbDFiEMyh71RJSy1m","object":"chat.completion.chunk","created":1687780896,"model":"gpt-35-turbo","choices":[{"index":0,"finish_reason":null,"delta":{"role":"assistant"}}],"usage":null}
+                 
+                 data:   {"id":"chatcmpl-7VfCSOSOS1gYQbDFiEMyh71RJSy1m","object":"chat.completion.chunk","created":1687780896,"model":"gpt-35-turbo","choices":[{"index":0,"finish_reason":null,"delta":{"content":"this"}}],"usage":null}
+                 
+                 data: {"id":"chatcmpl-7VfCSOSOS1gYQbDFiEMyh71RJSy1m","object":"chat.completion.chunk","created":1687780896,"model":"gpt-35-turbo","choices":[{"index":0,"finish_reason":null,"delta":{"content":" is "}}],"usage":null}
+                 
+                 
+                 
+                 data: {"id":"chatcmpl-7VfCSOSOS1gYQbDFiEMyh71RJSy1m","object":"chat.completion.chunk","created":1687780896,"model":"gpt-35-turbo","choices":[{"index":0,"finish_reason":null,"delta":{"content":"a text"}}],"usage":null}
```
Comment (adubovik) [inline src/test/java/com/epam/aidial/core/util/ModelCostCalculatorTest.java:181]:
```
data: {"error": { "message": "message", "type": "type", "param": "param", "code": "code" } } 
```

File: src/test/java/com/epam/aidial/core/util/ModelCostCalculatorTest.java [hunk 3]
```diff
@@ -61,4 +63,255 @@ public void testCalculate_TokenCost() {
...
+                    }
+                  ],
+                  "max_tokens": 500,
+                  "temperature": 1,
+                  "stream": false
+                }
+                """;
+        when(context.getRequestBody()).thenReturn(Buffer.buffer(request));
+
+        assertEquals(new BigDecimal("13.0"), ModelCostCalculator.calculate(context));
```
Comment (adubovik) [inline src/test/java/com/epam/aidial/core/util/ModelCostCalculatorTest.java:120]: Could you double check the final cost? Here is my calculations:

```
char_without_whitespace("A file is a named collection.") = 24

char_without_whitespace("How are you?") = 10

cost = 24 * 0.1 + 10 * 0.5 = 7.4
```

File: src/test/java/com/epam/aidial/core/util/ModelCostCalculatorTest.java (hunk 1 above)
Comment (astsiapanay) [reply src/test/java/com/epam/aidial/core/util/ModelCostCalculatorTest.java:134]: fixed

File: src/test/java/com/epam/aidial/core/util/ModelCostCalculatorTest.java (hunk 2 above)
Comment (astsiapanay) [reply src/test/java/com/epam/aidial/core/util/ModelCostCalculatorTest.java:181]: fixed

**Review by adubovik — APPROVED** [review]

**Pull request merged**